serde_json_path_core/src/spec/integer.rs

```diff
@@ -52,32 +52,30 @@ impl Integer {
         Self::try_new(value).expect("value is out of the valid range")
     }
 
-    /// Take the absolute value, producing `None` if the resulting value is outside
-    /// the valid range [-(2<sup>53</sup>)+1, (2<sup>53</sup>)-1].
-    pub fn checked_abs(mut self) -> Option<Self> {
-        self.0 = self.0.checked_abs()?;
-        self.check().then_some(self)
+    /// Take the absolute value, producing a new instance of [`Integer`]
+    pub fn abs(&self) -> Self {
```

Comment:
For all these methods I am not sure if for a type which is `Copy` it is more idiomatic to take `self` instead of `&self` or if there is some convention for it. Or if it even makes a difference in the end.

Rust's `std::num::NonZero` seems to use `self`. But maybe even in the standard library it is not consistent, see https://users.rust-lang.org/t/guidelines-for-self-ownership-on-copy-types/61262

Reply:
Yeah, I was a bit confounded by this myself. Originally I was just following the signature of [similar methods](https://doc.rust-lang.org/std/primitive.i64.html#method.checked_add), e.g., on `i64`, which take `self`, but thought `&self` made more sense in this PR. The link you provided is very helpful though, and I think that perhaps using `self` is the way to go, since `Integer` is just an `i64`.